ote/src/cljs/ote/views/transport_service_common.cljs

```diff
@@ -313,65 +312,86 @@
      :type          :checkbox}))
 
 
-(defn contact-info-group []
-  (form/group
-   {:label  (tr [:passenger-transportation-page :header-contact-details])
-    :columns 3
-    :layout :row}
-
-   (form/info (tr [:form-help :description-why-contact-info]))
-
-   {:name        ::common/street
-    :type        :string
-    :container-class "col-xs-12 col-sm-6 col-md-4"
-    :full-width?  true
-    :read (comp ::common/street ::t-service/contact-address)
-    :write (fn [data street]
-             (assoc-in data [::t-service/contact-address ::common/street] street))
-    :label (tr [:field-labels ::common/street])
-    :required? true
-    :max-length 128}
-
-   {:name        ::common/postal_code
-    :type        :string
-    :container-class "col-xs-12 col-sm-6 col-md-2"
-    :full-width?  true
-    :regex #"\d{0,5}"
-    :read (comp ::common/postal_code ::t-service/contact-address)
-    :write (fn [data postal-code]
-             (assoc-in data [::t-service/contact-address ::common/postal_code] postal-code))
-    :label (tr [:field-labels ::common/postal_code])
-    :required? true
-    :validate [[:postal-code]]}
-
-   {:name        ::common/post_office
-    :type        :string
-    :container-class "col-xs-12 col-sm-6 col-md-5"
-    :full-width?  true
-    :read (comp ::common/post_office ::t-service/contact-address)
-    :write (fn [data post-office]
-             (assoc-in data [::t-service/contact-address ::common/post_office] post-office))
-    :label (tr [:field-labels ::common/post_office])
-    :required? true
-    :max-length 64}
-
-   {:name        ::t-service/contact-email
-    :type        :string
-    :container-class "col-xs-12 col-sm-6 col-md-4"
-    :full-width?  true
-    :max-length 200}
+(defn contact-info-group [service]
+  (let [key (t-service/service-key-by-type (::t-service/type service))
+        service-sub-data (get service key)
+        contact-details-missing? (if (or
+                                       ;; contact details are not missing if address is given
+                                       (and
+                                         (not (empty? (get-in service-sub-data [::t-service/contact-address ::common/street])))
+                                         (not (empty? (get-in service-sub-data [::t-service/contact-address ::common/post_office])))
+                                         (and
```

Comment:
This `and` could be omitted and move children to parent `and`

Reply:
I'll change this later. And merge this now. Thank you for pointing this out.